src/app/components/product-table/product-table.component.html

```diff
@@ -1,28 +1,55 @@
 <div class="product-table mat-elevation-z4">
+
+
   <div *ngIf="loading" class="loading-shade">
     <mat-spinner></mat-spinner>
   </div>
   <table mat-table matSort [dataSource]="dataSource">
     <ng-container matColumnDef="name">
       <th mat-header-cell *matHeaderCellDef mat-sort-header> Name </th>
-      <td mat-cell *matCellDef="let row">{{row.name}}</td>
+      <td mat-cell *matCellDef="let row">{{ row.name }}</td>
     </ng-container>
 
     <ng-container matColumnDef="description">
       <th mat-header-cell *matHeaderCellDef mat-sort-header> Description </th>
-      <td mat-cell *matCellDef="let row">{{row.description}}</td>
+      <td mat-cell *matCellDef="let row" class="description"
+        ><p
+          #tooltip
+          [matTooltip]="row.description"
+          [matTooltipDisabled]="!isEllipsisActive(tooltip)"
+          matTooltipPosition="above"
+          aria-label="Hover for the whole text of description"
+          data-testid="description-text"
+        >
+          {{ row.description }}</p
+        ></td
+      >
     </ng-container>
 
     <ng-container matColumnDef="picture">
       <th mat-header-cell *matHeaderCellDef mat-sort-header> Picture Url </th>
-      <td mat-cell *matCellDef="let row" class="break-word">{{row.picture}}</td>
+      <td mat-cell *matCellDef="let row">
+        <div class="picture-url-container">
+          <button
+            mat-icon-button
+            aria-label="Copy Image URL"
+            [cdkCopyToClipboard]="row.picture"
+            data-testid="copy-picture-url"
```

Comment:
we can't mark elements using some custom attributes just for unit test purposes. 
find a way to locate this element in some other way.
remove this data-attr* (applies same in other places)